jukepoot/puppet_parser.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+"""A parser of /var/log/puppet.log.
+
+
+"""
+
+
+import argparse 
+from datetime import datetime
+import dateutil.parser
+import re
+import collections
+import os
+
+
+def main():
+	parser = argparse.ArgumentParser()
+	parser.add_argument(
+		help="parse the given filename", type=str, 
+		dest="filename", action="store")
+	parser.add_argument(
+		"-y","--year", help="parse file by setting start year", 
+		type=int, dest="year", action="store")
+	args = parser.parse_args()
+	if args.year:
+		print("year to use: {0:d}".format(args.year))
+		year = args.year
+
+	else: #build year for date parsing
+		year = datetime.utcfromtimestamp(
+			os.stat(args.filename).st_birthtime).year
+		# print("creation of file: {0:d}".format(year))
+
+	puppet_parser(args.filename,year)
+
+
+def puppet_parser(filepath, year):
+	"""Accepts the puppet file path and parses it out by calling other functions.
+
+	"""
+
+	# pattern to match from log file and group each finding for referencing
+	# pattern looks for a time foormat (x:x:x)any text after which is
+	# followed by the process and so on 
+	pattern = re.compile(
+		r'(^.+:..:..)\s(\S+)\s(\S+)(\[\d+\]):\s(.+$)')
+	#group 1 = date
+	#group 2 = hostname
+	#group 3 = proccess
+	#group 4 = PID
+	#group 5 = message
+	
+	# add to list for counter purposes 
+	message_list = []
+	proccess_list =[]
+	hostname_list= []
+	processID_list = []
+	
+	looped = 0
+	with open(filepath, "rU") as f:
+		for line in f:
+			if not pattern.match(line):
+				continue
+ 
+			current_date = pattern.match(line).group(1)
+			if looped >= 1:
```

Comment:
instead of having this "special" processing here I would just opt to have something like 

```previous_month```

variable that you first set to the None or something like that, and you can check if it is not none, then you can compare....